tests/helpers/main.cu

```diff
@@ -22,16 +23,9 @@ GTEST_API_ int main(int argc, char **argv) {
     printf("Running main() from %s\n", __FILE__);
     testing::InitGoogleTest(&argc, argv);
     auto rtn = RUN_ALL_TESTS();
-    // Reset all cuda devices for memcheck / profiling purposes.
-    int devices = 0;
-    gpuErrchk(cudaGetDeviceCount(&devices));
-    if (devices > 0) {
-        for (int device = 0; device < devices; ++device) {
-            gpuErrchk(cudaSetDevice(device));
-            gpuErrchk(cudaDeviceReset());
-        }
-    }
-    // If there were more than 1 tests selected, (to exlcude bad filters and FLAMEGPU_USE_GTEST_DISCOVER related spam)
+    // Reset all cuda devices for memcheck / profiling purposes (and finalize MPI)
+    flamegpu::util::cleanup();
```

Comment:
Given there are tests which call cleanup, will this work if there are MPI tests followed by cleanup tests?

Reply:
I think it works so long as cleanup is not called between MPI init and the last MPI test (cant call mpi finalise before mpi init which is testable). With them all being in the same Suite, their execution should be contiguous.

Reply:
Not a problem anymore with the split test suite (assuming no one runs the main test suite with mpirun).